synapse/rest/client/read_marker.py

```diff
@@ -50,17 +49,21 @@ async def on_POST(
 
         body = parse_json_object_from_request(request)
 
-        valid_receipt_types = {ReceiptTypes.READ, ReceiptTypes.FULLY_READ}
-        if self.config.experimental.msc2285_enabled:
-            valid_receipt_types.add(ReceiptTypes.READ_PRIVATE)
-
-        if set(body.keys()) > valid_receipt_types:
-            raise SynapseError(
-                400,
-                "Receipt type must be 'm.read', 'org.matrix.msc2285.read.private' or 'm.fully_read'"
-                if self.config.experimental.msc2285_enabled
-                else "Receipt type must be 'm.read' or 'm.fully_read'",
-            )
+        valid_receipt_types = {
+            ReceiptTypes.READ,
+            ReceiptTypes.FULLY_READ,
+            ReceiptTypes.READ_PRIVATE,
+        }
```

Comment:
I think you still want to consider READ_PRIVATE valid only if the flag is on?

Reply:
See https://github.com/matrix-org/synapse/pull/12670#discussion_r867819511. My understanding from what Erik said was that this only governs the warning we show, not the enforcement logic later(?)

Reply:
Yes, but would still be a confusing/incorrect warning.

Reply:
Although I guess the alternative is log spam...so maybe this is best.